File: clinic/src/patients/dto/create-patient.dto.ts
import { ApiProperty } from '@nestjs/swagger';
import {
  IsDateString,
  IsEnum,
  IsNotEmpty,
  IsOptional,
  IsString,
  MaxLength,
} from 'class-validator';
import { Gender, PatientStatus } from '../enums/patient.enum';

export class CreatePatientDto {
  @ApiProperty({
    description: 'Nombre(s) del paciente',
    example: 'Juan Carlos',
    maxLength: 100,
  })
  @IsString()
  @IsNotEmpty({ message: 'El nombre es obligatorio' })
  @MaxLength(100, { message: 'El nombre no puede exceder 100 caracteres' })
  firstName: string;

  @ApiProperty({
    description: 'Apellido(s) del paciente',
    example: 'Pérez García',
    maxLength: 100,
  })
  @IsString()
  @IsNotEmpty({ message: 'El apellido es obligatorio' })
  @MaxLength(100, { message: 'El apellido no puede exceder 100 caracteres' })
  lastName: string;

  @ApiProperty({
    description: 'Fecha de nacimiento del paciente (formato ISO 8601)',
    example: '1985-06-15',
    type: String,
  })
  @IsDateString(
    {},
    { message: 'La fecha debe estar en formato válido (YYYY-MM-DD)' },
  )
  birthDate: string;

  @ApiProperty({
    description: 'Género del paciente',
    enum: Gender,
    enumName: 'Gender',
    example: Gender.MALE,
  })
  @IsEnum(Gender, { message: 'El género debe ser M o F' })
  gender: Gender;

  @ApiProperty({
    description: 'Estado actual del paciente',
    enum: PatientStatus,
    enumName: 'PatientStatus',
    example: PatientStatus.ACTIVE,
    required: false,
    default: PatientStatus.ACTIVE,
  })
  @IsOptional()
  @IsEnum(PatientStatus, {
    message: 'El estado debe ser Activo, Seguimiento o Desactivado',
  })
  status?: PatientStatus;
}

File: auth-gateway/src/main/java/com/genosentinel/auth_gateway/dto/nestjs/CreatePatientDto.java
package com.genosentinel.auth_gateway.dto.nestjs;

import io.swagger.v3.oas.annotations.media.Schema;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Positive;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Schema(description = "Datos para crear un paciente")
public class CreatePatientDto {

    @NotBlank(message = "El nombre es obligatorio")
    @Schema(description = "Nombre completo del paciente", example = "Juan Pérez")
    private String firstName;

    @NotBlank(message = "Los apellidos son obligatorios")
    @Schema(description = "Apellidos del paciente", example = "García López")
    private String lastName;

    @NotBlank(message = "La fecha de nacimiento es obligatoria")
    @Schema(description = "Fecha de nacimiento del paciente en formato AAAA-MM-DD", example = "1980-05-15")
    private String birthDate;

    @NotBlank(message = "El género es obligatorio")
    @Schema(description = "Género del paciente", example = "M", allowableValues = {"M", "F", "Otro"})
    private String gender;

    @NotBlank(message = "El estado es obligatorio")
    @Schema(description = "Estado del paciente", example = "Activo", allowableValues = {"Activo", "Inactivo"})
    private String status;
}